Assets/General/Scripts/BoatLayout/MapShifting.cs

```diff
@@ -0,0 +1,85 @@
+/*****************************************************************************
+// File Name :         MapShifting.cs
+// Author :            Mark Hanson
+// Creation Date :     11/7/2024
+//
+// Brief Description : The array of different boat layouts that are accessible in other scripts
+*****************************************************************************/
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class MapShifting : MonoBehaviour
+{
+    //A way of adding layouts into the scene outside of this script
+    [Tooltip("Add all layouts of boat inside of here")]
+    [SerializeField]private protected GameObject[] _listOfLayouts;
+    private protected int _currentLayout = 0;
+    //For calling outside of script into another script
+    private UnityEvent _onMapShifting = new();
+    private UnityEvent _onMapReset = new();
+    /// <summary>
+    /// Allow for first layout to turn on if not on already
+    /// </summary>
+    void Start()
+    {
+        _listOfLayouts[_currentLayout].SetActive(true);
+        Subscribe();
+    }
+    /// <summary>
+    /// Turns off current layout of boat and goes to the next one
+    /// </summary>
+    private protected void LayoutSwapForward()
```

Comment:
If you have a swap forward you might want to include a swap backwards. Not certain if this is needed as I don't know if we would really ever NEED that functionality but more options are nice. Just a suggestion, not needed

Reply:
funny thing is I had one originally but I removed it for some reason